### ISSUE TITLE
Programme/Event Series/Exhibition relationships [migrated from old Collection repo]

### ISSUE THREAD

**Issue by jennpb** [wellcometrust/wellcomecollection.org#825]
@debdoop commented on [Tue Nov 17 2015](https://github.com/wellcometrust/collection/issues/21)

**Update**: the description here does not communicate purely a vision/rationale.  There are some answers in the comments.  David has recommended that Russell and a BE dev meet to discuss and write AC together.  There might be things here that need separate ticket.
## Description/vision/rationale

Quite often our events/artefacts may relate to more than one exhibition/event series/etc. And it would be useful to be able to make this clear, instead of trying to pick the most “important” one. We put something together last year to feed into designing the site, showing the complicated and layered relationships between various elements of our programme which can be referred to.

Also, can an "Event series" be related to an "Exhibition"? If not, we can use an in-line link. But in that case, can an "Event" be related to both an "Exhibition" and an "Event series"?

An “Event Series” (e.g. Why Music?) doesn’t show up on the Events page. It does appear in the calendar, but not the Events page and presumably it won’t get pulled through to the What’s On page either?

When we add an "Event Series”, we have to add it as an exhibition and then tag it as an event series. If we list the dates of the event series (which we often don’t need to as they’re ongoing), they then appear in the general Exhibitions section. It would be amazing if they could instead appear in the general Events section instead.

We have a weekend event called Why Music? which features many individual events, so it would be great to put the dates in (at the moment it’s in the copy to prevent it being listed under exhibitions).

The Hubbub Friday Spectacular event is listed as part of the Friday Spectacular event series. But it’s also part of the Hubbub event series (which I’ve added it to manually). We can choose only one to assign it to, despite both being appropriate.
## 

For an event series to not be categorised in the back end as an exhibition (eg. https://wellcomecollection.org/exhibitions/perspective-tours would become https://wellcomecollection.org/perspective-tours
## Q&A

NA
## Acceptance criteria

AC is TBC it does not capture everything discussed in the description above - suggest its moved to the AC:
- [ ] An individual event can be associated with any series and multiple series 
- [ ] An event series can be associated to an exhibition - and vice versa
- [ ] The multiple relationships are shown adjacent to 'Part of' on i) Listings, ii) event pages, as per the mock ups below
## Related/dependencies

NA
## Sketches/mockups/screenshots

![screen shot 2016-05-16 at 14 22 43](https://cloud.githubusercontent.com/assets/8453924/15290642/c1d239f4-1b71-11e6-8f8b-30ccbb8a1b63.png)

---

@debdoop commented on [Thu Apr 28 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-215418718)

will effect the listing and the related content and on the page it's self 
@drmwellcome will think about "part of..." section

---

@davidmillerwellcome commented on [Tue May 10 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-217154652)

Some questions...I think most are for @RussellDornan:

**1** Is this the doc referred to in the first paragraph [Events, Exhibitions & Tours Data Modelling](https://drive.google.com/open?id=0B7c9GFIDZ8nQTGZ4V01PVXF3c2c)?

**2** Not sure I understand the following...these look like question for back end dev, but we need to understand all the impacts on the site, and its not entirely clear what you're after in the second question:

> i Also, can an "Event series" be related to an "Exhibition"? If not, we can use an in-line link. But in that case, can an "Event" be related to both an "Exhibition" and an "Event series"?
> ii An “Event Series” (e.g. Why Music?) doesn’t show up on the Events page. It does appear in the calendar, but not the Events page and presumably it won’t get pulled through to the What’s On page either?

**3** I'm wondering if this should be pulled out as a separate ticket?  It seems like its not directly related, and might have separate FE/design work?:

> We have a weekend event called Why Music? which features many individual events, so it would be great to put the dates in (at the moment it’s in the copy to prevent it being listed under exhibitions).

**4** As an example for me to think about, is it correct to say you'd like [AD Tour of THIS IS A VOICE](http://wellcomecollection.org/events/ad-tour-voice) to be part of
i [THIS IS A VOICE](http://wellcomecollection.org/thisisavoice) AND
ii [Access Tours and Discussions](http://wellcomecollection.org/accesstours)

---

@RussellDornan commented on [Thu May 05 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-217158370)

@drmwellcome This is deffo best discussed in person...

---

@davidmillerwellcome commented on [Thu May 05 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-217160079)

@RussellDornan Can you answer the last point?  I can then do something thinking around that. 

I don't think we'll need 30 mins for accessibility tomorrow so we can discuss then.  I'll see if Dipak or Mike can come for the first half to discuss this ticket

---

@RussellDornan commented on [Thu May 05 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-217166088)

@drmwellcome. Hold on to your brain, because I'm about to take it for a ride 😮 

Point 1: Yes, the document you mention is the one, although I can see now that it's more complex than that lets on in terms of relationships. There are more connections.

Point 2 i: See Point 4.

Point 2 ii: On the Events page, it only lists single events so Event Series don't get pulled through on the Events page or the What's On page, which means people looking in those places won't see the main, overall event; only the single events that make up the series. Maybe there could be a way to choose whether an Event Series is to be listed on the Events page AND/OR the associated events? Separate tickie?

Point 3: An Event Series is (from a backend perspective) in the same category as an Exhibition. If we give an Event Series a date range at the moment, it then lists it under exhibitions. But we'd rather a) give it a date range and b) list it under Events (see Point 2 ii). Separate tickie here too?

Point 4: yes, that's correct.

Anything else I can tell you to yo' face! 😸 

---

@debdoop commented on [Fri May 06 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-217428904)

@drmwellcome  will sketch the primary relationships and the other relationships approach, where it will appear in different places on the site. 

---

@davidmillerwellcome commented on [Fri May 06 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-217437123)

Also agreed to have a meeting with me, @debdoop & @RussellDornan + assigned BE to discuss way forward.  Assigning to Debbie because I feel like the meeting is more likely to hold this up than my work!

---

@davidmillerwellcome commented on [Wed May 11 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-218408305)

Mockups shared with group on Slack for feedback.  Assigning to @josh to review.

---

@davidmillerwellcome commented on [Wed May 11 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-218420427)

The description here does not communicate purely a vision/rationale.  There are some answers in the comments.  Recommend that Russell and a BE dev meet to discuss and write AC together.  There might be things in the description that require separate tickets.

---

@debdoop commented on [Thu May 12 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-218745813)

Hey @RussellDornan and @drmwellcome I've added 

#160 an event series can't have a date range without becoming an exhibition listing
and 
#159 An event series doesn't show on the listings page

feel free to update and Russell to put in the backlog 

---

@RussellDornan commented on [Mon May 16 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-219393928)

@dannybirchall Hi Danny, can you give this a look so we can haz a chitty chat about it? 😎 

---

@RussellDornan commented on [Mon May 16 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-219420530)

Thanks, @Joshua-David 

I prefer the bold. I think I also prefer it without the colon. Thanks! 👍 

---

@Joshua-David commented on [Mon May 16 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-219423578)

@Thomas-Wellcome signed-off design for related tags below, assigning to you for FE

![screen shot 2016-05-16 at 14 22 43](https://cloud.githubusercontent.com/assets/8453924/15290642/c1d239f4-1b71-11e6-8f8b-30ccbb8a1b63.png)

---

@davidmillerwellcome commented on [Mon May 16 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-219441894)

@Thomas-Wellcome I just wanted to add, now that the designs have been signed off...I noticed that 'Part of' **and** the text that follows are part of the h2 heading.  
![screen shot 2016-05-16 at 15 35 06](https://cloud.githubusercontent.com/assets/14291978/15292694/2fc0336c-1b7c-11e6-951c-7f179c5a3df8.png)

It breaks the purpose of a heading to have the content that follows inside the heading tag.  So I think it should be:

```
<h2>Part of</h2>
Access Tours...
```

If you agree, we should be update the AC.

---

@Joshua-David commented on [Mon May 23 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-220937584)

@nambisolo html to replace the current .snippet__related-link:

```
<div class="snippet__related-link">
  <h4>Part of</h4>
  <ul>
    <li><a href="/collection/www/thisisavoice">Title of related content</a></li>
    <li><a href="/collection/www/thisisavoice">Title of related content</a></li>
    <li><a href="/collection/www/thisisavoice">Title of related content</a></li>
  </ul>
</div>
```

---

@nambisolo commented on [Mon May 23 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221024222)

@Joshua-David see http://wellcomecollection.org/events/symposium-out-control. Multiple associations will be listed at the bottom of this page too. Can you tweak the markup for this as well please?

---

@Joshua-David commented on [Tue May 24 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221220521)

@nambisolo:

```
<div class="snippet__related-link">
  <h2>Part of</h2>
  <h2>
    <ul>
     <li><a href="/collection/www/thisisavoice">Title of related content</a></li>
     <li><a href="/collection/www/thisisavoice">Title of related content</a></li>
     <li><a href="/collection/www/thisisavoice">Title of related content</a></li>
    </ul>
  </h2>
</div>
```

---

@nambisolo commented on [Tue May 24 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221322971)

Books/Publications are adding some complexity to this. Can't add 'Part of' the same way as other items. Publications were a type of Artefact. Books are a content type rather than a type of Artefact. Attached is a query returning all 'Publications' on the site.

[field_artefact_type.xlsx](https://github.com/wellcometrust/collection/files/279933/field_artefact_type.xlsx)

---

@RussellDornan commented on [Tue May 24 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221325828)

Hi @nambisolo 

You can ignore Publications, as these will soon be removed once the bugs with Books have been fixed. Are Books still a problem?

---

@nambisolo commented on [Tue May 24 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221326387)

@RussellDornan I think fixing the bugs with the books should be done before this ticket.

---

@debdoop commented on [Wed May 25 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221559190)

Russell is happy to go ahead without books and we can sort them out in the future so he won't link to them from this section :ok_hand: 

---

@nambisolo commented on [Wed May 25 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221592422)

@RussellDornan I think I've managed to untangle some things. I'll detail below with some questions but it may be easier to discuss in person.

**Books/Publications**
I discovered that as books could not be added as a 'Referenced content listing item' to an exhibition, they were being added as a 'Custom listing item'. This is an obvious workaround but it was preventing 'Part of' being displayed. I have enabled books to be referenced now and 'Part of' can be displayed under books in listings.

Unfortunately this requires retrospective editing of all pages that have been referencing books like this. Be careful when carrying out the editing as most books will have a corresponding 'publication'. Make sure the 'book' content type and not the 'artefact' content type is added.

Additionally to ensure the book snippet is displayed correctly on listing pages I've had to remove some markup. This has had a minor knock on effect on the books landing page which will need some FE attention.

**Rendering associations on each content type**
The only content type that currently displays associations automatically is 'event'. Each of the following can do so as well with coding but I need advice on the positioning:
- Artefact
- Basic page
- Book
- Exhibition

To overcome the limitation of the above content types to automatically display associations, you've been adding them to the wysiwyg field. Again, understandable but this copy will need to be removed in order to avoid duplication when associations are drawn in automatically.

---

@nambisolo commented on [Thu May 26 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221918140)

@RussellDornan I have added the 'Part of' field to the foot of the 'Artefact' content type but I've noticed items are already being automatically generated at the bottom of the page. From what I can see, these items are Exhibitions from which the artefacts are referenced via the 'Artefacts' field.

If you want to use the 'Part of' field on Artefacts pages I suggest you remove the artefacts references from the Exhibition and add them to the 'Part of' field instead. Over time the Artefacts field will become redundant.

Let me know if this makes sense.

---

@RussellDornan commented on [Thu May 26 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-221926404)

@nambisolo That makes perfect sense (and is what I tried to explain at your desk previously, though you've done it much more articulately here). All is crystal clear!

---

@nambisolo commented on [Mon Jun 13 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-222091852)

@Joshua-David I've added the markup to the BE code. I've also made some changes to the markup that renders the book snippets for consistency. As a result the alignment on the books landing page has changed from centred to left. Could you make the necessary CSS changes to realign please?

---

@Joshua-David commented on [Fri May 27 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-222138272)

@Thomas-Wellcome have assigned to you for code review

---

@Thomas-Wellcome commented on [Tue Jun 14 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-225880453)

@nambisolo I know you changed the markup for books, but without it the titles won't sit on same line. e.g. "A Guide for the Incurably Curious":

![](http://content.screencast.com/users/TTrusch/folders/Jing/media/df92fe44-9697-43cb-a278-0c08b6ffa3d5/00000242.png)

What trouble causes it if we reimplement the wrapper divs?

The missing div has had the class < div class="publications-list__cover" > and was removed in commit named "#21 Enabled 'book' content type to be referenced as an entity. Added markup for formatting in page associations.".

---

@Thomas-Wellcome commented on [Wed Jun 22 2016](https://github.com/wellcometrust/collection/issues/21#issuecomment-227794706)

@nambisolo Could you please reinsert the < div class="publications-list__cover" > wrapper for books.  I changed the CSS to only affect the design on book page, but not on search results. Thank you!

**Comment by jennpb** [comment]
@jamesgorrie @Heesoomoon ^^

Ahoy, some useful information about the events and exhibitions data model, from the V1 site repo.

**Comment by dannybirchall** [comment]
That's v3. You're working on v4. #historyman
